clients/drcachesim/tools/basic_counts.cpp

```diff
@@ -215,6 +234,12 @@ basic_counts_t::print_counters(const counters_t &counters, int_least64_t num_thr
     }
     std::cerr << std::setw(12) << counters.instrs_nofetch << prefix
               << " non-fetched instructions\n";
+    if (counters.is_counting_kernel_instrs()) {
```

Comment:
Instead of tracking is_counting_kernel_instrs, can we just pass a bool to print_counters `for_kernel_trace` (set by the caller based on the file type, or maybe stored as a var in basic_counts_t) and use it here?

Reply:
Done.